cfgov/unprocessed/apps/regulations3k/js/permalinks-utils.js

```diff
@@ -74,12 +82,125 @@ const getCurrentParagraph = ( currentPosition, paragraphs ) => {
+  if ( commentParagraph !== '' ) {
+    commentParagraph = '-' + commentParagraph;
+  }
+
+  return commentedSection + commentedParagraph + commentParagraph;
+};
+
+/**
+ * getWayfinderInfo - process paragraph to create wayfinder
+ * @param {string} label - label of current paragraph
+ * @param {string} sectionTitle - title of current section
+ *
+ * @returns {object} object of the values for wayfinder
+ */
+const getWayfinderInfo = ( label, sectionTitle ) => {
+  let sectionFormattedTitle;
+  let paragraphMarker;
+
+  // For interpretations, the wayfinder should look like "Comment 4(a)-1.iv.A"
+  // Or like "Comment app. G-1.iv.A" for interpretations of appendices
+  if ( sectionTitle.indexOf( 'Comment for ' ) > -1 || label.indexOf('Interp') > -1 ) {
+    sectionFormattedTitle = 'Comment ';
+    paragraphMarker = getCommentMarker( label );
+  } else if ( sectionTitle.indexOf( 'Appendix ' ) === 0 ) {
+    // For appendices, the wayfinder should look like "Appendix A"
+    sectionFormattedTitle = sectionTitle.match( wayfinderRegex.appendixTitle )[0];
+    paragraphMarker = '';
+  } else {
+    // For sections of the main regulation text, the wayfinder should look like "§ 1026.5(b)(2)(ii)(A)(1)""
+    sectionFormattedTitle = sectionTitle.match( wayfinderRegex.title )[0];
+    if ( label !== '' ) {
+      paragraphMarker = '(' + label.replace( wayfinderRegex.marker, ')(' ) + ')';
+    }
+  }
+
+  return {
+    paragraphMarker: paragraphMarker,
+    formattedTitle: sectionFormattedTitle
+  };
+};
+
+/**
+ * updateWayfinder - Update the Wayfinder element with current paragraph info
+ * @param {Boolean} scroll - if true, the function will scroll the current paragraph into view
+ * @param {object} wayfinder - wayfinder HTML element
+ * @param {object} mainContent - regs main section HTML element
+ */
+const updateWayfinder = function( scroll, wayfinder, mainContent ) {
+  // We can usually rely on the constants declared above, but just in case...
+  if ( typeof wayfinder === 'undefined' ) wayfinder = regs3kWayfinder;
+  if ( typeof mainContent === 'undefined' ) mainContent = regs3kMainContent;
+
+  if ( wayfinder !== null && mainContent !== null ) {
+    let paragraphMarker;
+    let sectionFormattedTitle;
+    let wayfinderInfo;
+    const wayfinderLink = wayfinder.querySelector( '.o-regulations-wayfinder_link' );
+    const currentParagraph = getCurrentParagraph( scrollY() + wayfinderOffset, paragraphPositions );
+
+    if ( currentParagraph ) {
+      // To avoid hashed IDs, we use data-label for wayfinder text formatting
+      const currentLabel = document.getElementById( currentParagraph ).dataset.label;
+      const sectionTitle = wayfinder.dataset.section;
+      wayfinderInfo = getWayfinderInfo( currentLabel, sectionTitle );
+      paragraphMarker = wayfinderInfo.paragraphMarker;
+      sectionFormattedTitle = wayfinderInfo.formattedTitle;
+      mainContent.classList.add( 'show-wayfinder' );
+      wayfinderLink.href = '#' + currentParagraph;
```

Comment:
Clicking the wayfinder's link positions the current paragraph at the top of the viewport which is behind the wayfinder. Ideally, it'd do something similar to what's on L200 and position it 60 pixels higher.

<img src="https://user-images.githubusercontent.com/1060248/56631572-df6f1b80-6623-11e9-9a90-4decfbcc74cd.png" alt="wayfinder" width="300">